platform/web/jsonschema.go

```diff
@@ -0,0 +1,92 @@
+package web
+
+import (
+	"fmt"
+	"github.com/fewlinesco/gojsonschema"
+)
+
+// JSONSchemaAdditionalPropertyError is returned when the validation failed due to additional properties being present in the JSON input.
+type JSONSchemaAdditionalPropertyError struct {
+	Details ErrorDetails
+}
+
+func (err JSONSchemaAdditionalPropertyError) Error() string {
+	return "the json input contains unknown keys"
+}
+
+// JSONSchemaMissingPropertyError is returned when the validation failed due to missing required properties in the JSON input.
+type JSONSchemaMissingPropertyError struct {
+	Details ErrorDetails
+}
+
+func (err JSONSchemaMissingPropertyError) Error() string {
+	return "the json input is missing required keys"
+}
+
+// JSONSchemaValidationError is returned when the validation failed
+type JSONSchemaValidationError struct {
+	Details ErrorDetails
+}
+
+func (err JSONSchemaValidationError) Error() string {
+	return "the json input is not valid against the json schema"
+}
+
+// ValidateJSONAgainstSchema validates a JSON object against a JSON schema and returns validation errors if there are any.
+func ValidateJSONAgainstSchema(jsonData []byte, jsonSchema []byte) error {
+	return validateJSONAgainstSchema(jsonData, gojsonschema.NewBytesLoader(jsonSchema))
+}
```

Comment:
Why are you adding this "abstraction"?

Reply:
Actually it's the same reason I didn't used a separate package which I tried at some point. I wanted to preserve the current API to avoid breaking changes while using the new refactored components. But I'm think now I should leave the old functions untouched and just add the new functions we wanted, it'll be a lot simpler than adapting the existing functions we want to deprecate.